nix/master.nix

```diff
@@ -444,6 +444,99 @@ in
+            lib.types.submodule (
+              { name, ... }:
+              {
+                options = {
+                  defaultBranch = lib.mkOption {
+                    type = lib.types.str;
+                    description = ''
+                      The repositories default branch.
+                    '';
+                  };
+
+                  url = lib.mkOption {
+                    type = lib.types.str;
+                    description = ''
+                      The repository's URL, must be fetchable by git.
+                    '';
+                  };
+
+                  pollInterval = lib.mkOption {
+                    type = lib.types.addCheck lib.types.int (x: x > 0);
+                    default = 5 * 60;
+                    description = ''
+                      How often to poll this repository expressed in seconds.
+                    '';
+                  };
+
+                  # sshPrivateKeyFile = lib.mkOption {
+                  #   type = lib.types.nullOr lib.types.path;
+                  #   default = cfg.pullBased.sshPrivateKey;
+                  #   description = ''
+                  #     If non-null the specified SSH key will be used to fetch all configured repositories.
+                  #     This option is defaults to the global `sshPrivateKeyFile` option.
+                  #   '';
+                  # };
+
+                  # sshKnownHostsFile = lib.mkOption {
+                  #   type = lib.types.nullOr lib.types.path;
+                  #   default = cfg.pullBased.ssh_known_hosts;
+                  #   description = ''
+                  #     If non-null the specified known hosts file will be matched against when connecting to
+                  #     repositories over SSH. This option defaults to the global `sshKnownHostsFile` option.
+                  #   '';
+                  # };
+                };
+              }
+            )
+          );
+        };
+
+        pollInterval = lib.mkOption {
+          type = lib.types.addCheck lib.types.int (x: x > 0);
+          default = 5 * 60;
+          description = ''
+            How often to poll each repository by default expressed in seconds. This value can be overriden
+            per repository.
+          '';
+        };
+
+        pollSpread = lib.mkOption {
+          type = lib.types.nullOr lib.types.int;
+          default = null;
+          description = ''
+            If non-null and non-zero pulls will be randomly spread apart up to the specificied
+            number of seconds. Can be used to avoid a thundering herd situation.
+          '';
+        };
+
+        sshPrivateKeyFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          description = ''
+            If non-null the specified SSH key will be used to fetch all configured repositories.
+            This option is overriden by the per-repository `sshPrivateKeyFile` option.
```

Comment:
The other option seems to be not actually implemented. So we either remove it from the description or implement it.